camel/typing.py

```diff
@@ -24,14 +24,34 @@ class RoleType(Enum):
 
 class ModelType(Enum):
     GPT_3_5_TURBO = "gpt-3.5-turbo"
+    GPT_3_5_TURBO_16K = "gpt-3.5-turbo-16k"
     GPT_4 = "gpt-4"
     GPT_4_32k = "gpt-4-32k"
     STUB = "stub"
 
     @property
-    def value_for_tiktoken(self):
+    def value_for_tiktoken(self) -> str:
         return self.value if self.name != "STUB" else "gpt-3.5-turbo"
 
+    @property
+    def token_limit(self) -> int:
```

Comment:
I've been thinking to have this property scattered across the concrete classes. But okay, let this entity hang out here for a while.

Reply:
Do you mean having it in the `OpenAIModel`? We can add it.

Reply:
Updated. Can you check again?